pkg/reconciler/kubernetes/tektonchain/tektonchain.go

```diff
@@ -391,8 +394,9 @@ func (r *Reconciler) transform(ctx context.Context, manifest *mf.Manifest, comp
 		common.InjectOperandNameLabelOverwriteExisting(v1alpha1.OperandTektoncdChains),
 		common.DeploymentImages(chainImages),
 		common.AddConfiguration(instance.Spec.Config),
-		common.AddConfigMapValues(ChainsConfig, instance.Spec.Chain),
+		common.AddConfigMapValues(ChainsConfig, instance.Spec.Chain.ChainProperties),
 		common.AddDeploymentRestrictedPSA(),
+		AddMongoServerUrl(instance.Spec.Chain.Params),
```

Comment:
what to do with other key value pairs of param field?

Reply:
i agree, there are no other fields in param field right now - which should continue being the case
i feel that instead of a new params field, this can go in as a separate field that gets exposed as the env var inside the chains controller pod

Reply:
something like
```
spec:
  targetNamespace: tekton-pipelines
  mongoServerUrl: xxx://xxx:xxx
  artifacts.taskrun.format: in-toto
  artifacts.taskrun.storage: tekton,oci (comma separated values)
  artifacts.taskrun.signer: x509

```

Reply:
mongo server url might have password/token. Is it ok to keep it in TektonConfig CR?

Reply:
instead of this, should we add field like `envs` or `controllerEnvs` which is an array of key value pairs. whatever specified here will be added as env to chains controller.

adding a field which is getting added to controller or other field are getting added to configmap, this is not consistent, and give an wrong impression/UX